app/actions/handlers.py

```
@@ -203,6 +203,20 @@ async def action_pull_observations(integration, action_config: PullObservationsC
 
 async def process_data_file(file_name, integration, process_config):
     logger.info(f"Processing data file {file_name} for integration {integration}...")
+
+    # Check current file status
+    file_status = await get_file_status(file_name)
+
+    if not file_status:
+        logger.warning(f"File {file_name} not found. Skipping processing.")
+        return 0
+    if file_status.get("status") == FileStatus.IN_PROGRESS.value:
+        logger.warning(f"File {file_name} is already in progress. Skipping processing.")
+        return 0
+    if file_status.get("status") == FileStatus.PROCESSED.value:
+        logger.warning(f"File {file_name} had been processed. Skipping processing.")
+        return 0
+
```

I'd remove these from here to avoid a chance of hiding errors.